### PR TITLE
Closes #1125 Success and failure workflow handler.

### PR DESCRIPTION
**Pull request by MrK191**
Also merged with #1123.

### CODE REVIEW

File: src/test/java/pl/edu/icm/pl/mxrdr/extension/workflow/handler/MxrdrNotificationSenderTest.java [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import static edu.harvard.iq.dataverse.workflow.WorkflowContext.TriggerType.PostPublishDataset;
+import static java.util.Collections.emptyMap;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+
+@ExtendWith(MockitoExtension.class)
```
Comment (kaczynskid) [inline src/test/java/pl/edu/icm/pl/mxrdr/extension/workflow/handler/MxrdrNotificationSenderTest.java:41]: `MockitoExtension` is only needed when using `@Mock` annotation. When using direct calls to `Mockito.mock(..)` like in this class, which I prefer and recommend, nothing extra is neede.

File: src/test/java/pl/edu/icm/pl/mxrdr/extension/workflow/handler/MxrdrNotificationSenderTest.java [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+public class MxrdrNotificationSenderTest {
+
+    private MxrdrNotificationSender mxrdrNotificationSender;
+
+    private UserNotificationDao userNotificationDao;
+
+    private MailService mailService;
+
+    private AuthenticatedUser user;
+    private final Instant timer = Instant.ofEpochSecond(1592570489);
```
Comment (kaczynskid) [inline src/test/java/pl/edu/icm/pl/mxrdr/extension/workflow/handler/MxrdrNotificationSenderTest.java:51]: This `1592570489` is a magic number. Pleas use sth. like `Instant.parse("2020-06-01T09:10:20.00Z")` instead to make it readable. Or better yet `Clock clock = Clock.fixed(Instant.parse("2020-06-01T09:10:20.00Z"), UTC);`

File: src/test/java/pl/edu/icm/pl/mxrdr/extension/workflow/handler/MxrdrNotificationSenderTest.java (hunk 1 above)
Comment (MrK191) [reply src/test/java/pl/edu/icm/pl/mxrdr/extension/workflow/handler/MxrdrNotificationSenderTest.java:41]: Yeah at first I was using annotations, then decided to switch and forgot to remove it :P.

**Comment by madryk** [comment]
Can we clean up the mess with `src/main/resources/notificationTemplate.xhtml` and `src/main/webapp/pluggable/notifications.xhtml`?

**Comment by MrK191** [comment]
@madryk I removed notificationTemplate, I wasn't sure who should do what since this change was made by Darek.

**Pull request merged**